python/ray/dag/compiled_dag_node.py

```diff
@@ -277,9 +277,14 @@ def resolve(self, channel_results: Any) -> Any:
         """
 
         if isinstance(self.input_variant, ChannelInterface):
-            value = channel_results[self.channel_idx]
+            future = channel_results[self.channel_idx]
+            assert isinstance(future, DAGOperationFuture)
+            value = future.wait()
         else:
             value = self.input_variant
+        if isinstance(value, ReadyFuture):
+            temp = value.wait()
+            print(f"{temp=}, {future=}")
```

Comment:
I am a bit confused here why we need to call `.wait()` twice? Also, why do we need to check `isinstance(value, ReadyFuture)` instead of `DAGOperationFuture`?

Reply:
Seems better to put the `.wait()` in `reset_intermediate_buffer`?

Reply:
For the first question, L285-287 is my debugging code and has been removed.

Reply:
Fixed